core/domain/learner_progress_services.py

```diff
@@ -398,29 +454,34 @@ def get_all_completed_exp_ids(user_id):
         return []
 
 
-def get_completed_exp_summaries(user_id):
+def get_filtered_completed_exp_summaries(exploration_summaries,
+                                         exploration_ids):
     """Returns a list of summaries of the completed exploration ids and the
     number of explorations deleted from the list as they are no longer present.
 
     Args:
-        user_id: str. The id of the learner.
+        exploration_summaries: list(ExplorationSummary). The list of exploration
+            summary domain objects to be filtered.
+        exploration_ids: list(str). The ids of the explorations corresponding to
+            the exploration summary domain objects.
 
     Returns:
-        list(ExplorationSummary). List of ExplorationSummary domain objects of
-            the completed explorations.
-        int. The number of explorations deleted from the list as they are no
-            longer present.
+        tuple. A 2-tuple whose elements are as follows:
+        - list(ExplorationSummary). Filtered list of ExplorationSummary domain
+                objects of the completed explorations.
+        - list(str). The ids of the explorations that are no longer present.
     """
-    completed_exploration_ids = get_all_completed_exp_ids(user_id)
+    index = 0
+    nonexistent_completed_exp_ids = []
+    filtered_completed_exp_summaries = []
+    for exploration_summary in exploration_summaries:
```

Comment:
for index, exploration_summary in enumerate(exploration_summaries)

(and then you don't need to keep track of index separately)